bundles/org.openhab.binding.meteoalerte/src/main/java/org/openhab/binding/meteoalerte/internal/MeteoAlertIconProvider.java

```diff
@@ -0,0 +1,128 @@
+    private static final String DEFAULT_DESCRIPTION = "Icons illustrating weather events provided by Météo Alerte";
+    private static final Set<String> ICONS = Set.of(WAVE, AVALANCHE, HEAT, FREEZE, FLOOD, SNOW, STORM, RAIN, WIND,
+            "meteo_france");
+
+    private final Logger logger = LoggerFactory.getLogger(MeteoAlertIconProvider.class);
+    private final BundleContext context;
+    private final TranslationProvider i18nProvider;
+
+    @Activate
+    public MeteoAlertIconProvider(final BundleContext context, final @Reference TranslationProvider i18nProvider) {
+        this.context = context;
+        this.i18nProvider = i18nProvider;
+    }
+
+    @Override
+    public Set<IconSet> getIconSets() {
+        return getIconSets(null);
+    }
+
+    @Override
+    public Set<IconSet> getIconSets(@Nullable Locale locale) {
+        String label = getText("label", DEFAULT_LABEL, locale);
+        String description = getText("decription", DEFAULT_DESCRIPTION, locale);
+
+        return Set.of(new IconSet(BINDING_ID, label, description, Set.of(Format.SVG)));
+    }
+
+    private String getText(String entry, String defaultValue, @Nullable Locale locale) {
+        String text = defaultValue;
+        if (locale != null) {
+            text = i18nProvider.getText(context.getBundle(), "iconset." + entry, defaultValue, locale);
+            text = text == null ? defaultValue : text;
+        }
+        return text;
+    }
+
+    @Override
+    public @Nullable Integer hasIcon(String category, String iconSetId, Format format) {
+        return ((ICONS.contains(category) && iconSetId.equals(BINDING_ID))
+                || (ICONS.contains(category.replace(BINDING_ID + ":", "")))) && format == Format.SVG ? 7 : null;
+    }
+
+    public @Nullable InputStream getIcon(String category, String state) {
+        return getIcon(category, BINDING_ID, state, Format.SVG);
+    }
+
+    @Override
+    public @Nullable InputStream getIcon(String category, String iconSetId, @Nullable String state, Format format) {
+        String icon = getResource(category);
+        if (icon.isEmpty()) {
+            return null;
+        }
+
+        if (state != null) {
+            try {
+                Integer ordinal = Integer.valueOf(state);
+                AlertLevel alertLevel = ordinal < AlertLevel.values().length ? AlertLevel.values()[ordinal]
+                        : AlertLevel.UNKNOWN;
+                icon = icon.replaceAll(AlertLevel.UNKNOWN.color, alertLevel.color);
+
+            } catch (NumberFormatException e) {
+                logger.debug("{} is not a valid DecimalType", state);
+            }
+        }
+
+        return new ByteArrayInputStream(icon.getBytes());
+    }
+
+    private String getResource(String iconName) {
+        String result = "";
+
+        URL iconResource = context.getBundle()
+                .getEntry("icon/%s.svg".formatted(iconName.replace(BINDING_ID + ":", "")));
```

Comment:
Same here, `iconName.replace(BINDING_ID + ":", "")` is just a wrokaround that should be removed.